fdbclient/GlobalConfig.actor.cpp

```diff
@@ -182,12 +182,20 @@ ACTOR Future<Void> GlobalConfig::refresh(GlobalConfig* self) {
 	// TraceEvent trace(SevInfo, "GlobalConfig_Refresh");
 	self->erase(KeyRangeRef(""_sr, "\xff"_sr));
 
-	Transaction tr(self->cx);
-	tr.setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
-	RangeResult result = wait(tr.getRange(globalConfigDataKeys, CLIENT_KNOBS->TOO_MANY));
-	for (const auto& kv : result) {
-		KeyRef systemKey = kv.key.removePrefix(globalConfigKeysPrefix);
-		self->insert(systemKey, kv.value);
+	state Transaction tr(self->cx);
+	loop {
+		try {
+			tr.setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
```

Comment:
Add `tr.reset()` above this line. The transaction will also need to be reset in the above retry loop.

Reply:
`onError` should be resetting the transaction for us, unless there's some other issue here. It may be necessary to do a transaction refresh here, though, since we want to drop the reference to cx temporarily.

Reply:
Interesting, I saw a transaction_too_old error that was fixed by adding a reset at the top. I've also seen other places that use `onError` and `reset`, is that old behavior?

Reply:
It should not be necessary to add a reset at the top of a transaction loop when using `onError`. If you have two consecutive retry loops that use the same transaction, it is often necessary to reset the transaction in between. That would only need to be done once, though, not in each iteration of the second loop.

Occasionally there are also retry loops where we want the transaction to undergo a full reset so that it doesn't have the normal backoff behavior. That should be unusual, though.